src/test/HostActivationTests/GivenThatICareAboutNativeHostApis.cs

```diff
@@ -96,6 +90,142 @@ public void Breadcrumb_thread_does_not_finish_when_app_has_unhandled_exception()
+
+            var exeDir = Path.Combine(fixture.TestProject.ProjectDirectory, "ed");
+            var programFiles = Path.Combine(exeDir, "pf");
+            var workingDir= Path.Combine(fixture.TestProject.ProjectDirectory, "wd");
+            var globalSdkDir = Path.Combine(programFiles, "dotnet", "sdk");
+            var localSdkDir = Path.Combine(exeDir, "sdk");
+
+            // start with an empty global.json, it will be ignored, but prevent one lying on disk 
+            // on a given machine from impacting the test.
+            var globalJson = Path.Combine(workingDir, "global.json");
+            Directory.CreateDirectory(workingDir);
+            File.WriteAllText(globalJson, "{}");
+
+            foreach (string sdk in globalSdks)
+            {
+                Directory.CreateDirectory(Path.Combine(globalSdkDir, sdk));
+            }
+
+            foreach (string sdk in localSdks)
+            {
+                Directory.CreateDirectory(Path.Combine(localSdkDir, sdk));
+            }
+
+            var dotnetLocation = Path.Combine(dotnet.BinPath, $"dotnet{fixture.ExeExtension}");
+            string expectedList;
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) // multilevel lookup does not exist otherwise
+            {
+                // With multi-level lookup (windows onnly): get local and global sdks sorted by ascending version,
+                // with global sdk coming before local sdk when versions are equal
+                expectedList = string.Join(';', new[]
+                {
+                     Path.Combine(localSdkDir, "0.1.2"),
+                     Path.Combine(globalSdkDir, "1.2.3"),
+                     Path.Combine(localSdkDir, "1.2.3"),
+                     Path.Combine(globalSdkDir, "2.3.4-preview"), 
+                     Path.Combine(globalSdkDir, "4.5.6"),
+                     Path.Combine(localSdkDir, "5.6.7-preview"),
+                });
+                dotnet.Exec(appDll, new[] { "hostfxr_get_available_sdks", exeDir } )
+                    .EnvironmentVariable("TEST_MULTILEVEL_LOOKUP_PROGRAM_FILES", programFiles)
+                    .CaptureStdOut()
+                    .CaptureStdErr()
+                    .Execute()
+                    .Should()
+                    .Pass()
+                    .And
+                    .HaveStdOutContaining("hostfxr_get_available_sdks:Success")
+                    .And
+                    .HaveStdOutContaining($"hostfxr_get_available_sdks sdks:[{expectedList}]");
+            }
+
+            // Without multi-level lookup: get only sdks sorted by ascending version
+            expectedList = string.Join(';', new[]
+            {
+                 Path.Combine(localSdkDir, "0.1.2"),
+                 Path.Combine(localSdkDir, "1.2.3"),
+                 Path.Combine(localSdkDir, "5.6.7-preview"),
+            });
+
+            dotnet.Exec(appDll, new[] { "hostfxr_get_available_sdks", exeDir })
```

Comment:
Personally I would prefer to split this into multiple tests. The setup can be either shared, or rerun every time, it's not that expensive. I don't like tests which run the product multiple times - it's really hard to debug.

Reply:
You're right. I'll fix it.